recipes/full_finetune_distributed.py

```diff
@@ -563,6 +584,12 @@ def recipe_main(cfg: DictConfig) -> None:
         )
 
     init_process_group(backend="gloo" if cfg.device == "cpu" else "nccl")
+    if cfg.get("fsdp_cpu_offload", False):
+        # Utilize all available CPU cores for intra-op parallelism. This provides ~2x
+        # speed up when benchmarking fused AdamW on CPU
+        threads = os.cpu_count() // torch.distributed.get_world_size()
```

Comment:
How did you calculate this? I opened the draftiest PR https://github.com/pytorch/pytorch/pull/126199 to account for other environments where CPU affinity is set and cgroups could be used, and plan on talking with Intel to get this into better shape. But why `os.cpu_count() // torch.distributed.get_world_size()`?

Reply:
@janeyx99 Unfortuantely I don't have the bets answer to this, @msaroufim suggested following this over chat and referred to answerAI doing similar (following his suggestions) - https://github.com/AnswerDotAI/fsdp_qlora/commit/1a9fddfb660b85e46bdf18c3fcf0d599628c9c6c

Intuitively, this makes sense since it's essentially assigning each process an equal number of vCPUs out of the vCPUs available. But as you mentioned, this doesn't take into account physical CPUs being < vCPUs, cgroups, and the like. Mostly meant to be a stopgap for now, and I'm sure perf will be very different on different HW set ups. In the medium term, we might want to use what you end up landing in torch core, or even expose this to the user so they can configure this.

Reply:
Yeh I suspect comprehensive benchmarking with how diverse cpus are might take a while. So this is a fine stopgap